server/src-lib/Hasura/Db.hs

```diff
@@ -66,11 +66,12 @@ lazyTxToQTx = \case
 
 runLazyTx
   :: PGExecCtx
+  -> Maybe Q.TxAccess
```

Comment:
I get that the underlying API allows `Nothing`, which presumably means “use the value of the `default_transaction_read_only` configuration parameter, but it seems like it would be clearer to accept a `Q.TxAccess` directly here and just pass `Q.ReadWrite` instead of `Nothing`. That seems more correct anyway, since the places where we use `Nothing` currently need `Q.ReadWrite` access.